core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -205,6 +205,61 @@ class ReplicaManagerTest {
     when(cache.getAliveBrokerNodes(any[ListenerName])).thenReturn(aliveBrokers)
   }
 
+  @Test
+  def testMaybeAddLogDirFetchersForV1MessageFormat(): Unit = {
```

Comment:
You mentioned that the v2 format wasn't as easy to replicate. Is it too hard to have a test for? Or are the cases similar enough for v1 to cover both and v1 is less flaky?

Reply:
V1 is deprecated and we intend to remove it (along with V0). We should not be writing new tests with that.

Reply:
> You mentioned that the v2 format wasn't as easy to replicate.

The bug is happened when ALTER thread fetches offset according to V1 flow (as the epoch cache has not been updated). Hence, the test was used to verify the V1 flow.


> V1 is deprecated and we intend to remove it (along with V0). We should not be writing new tests with that.

ok, I will remove the epoch cache manually instead of using V1 configs.